src/track/track.cpp

```diff
@@ -216,6 +216,9 @@ void Track::importMetadata(
     // If the Serato tags are empty they are not present. In this case
     // all existing metadata must be preserved. Importing Serato metadata
     // is done last to override any previously imported metadata.
+    //
+    // TODO: Import Serato metadata within the locking scope and not
```

Comment:
@Holzhaus This could be done in a subsequent refactoring step.

For Serato empty tags importBeats() and importCueInfos() should return nullptr and getTrackColor() should return std::nullopt in this case (requiring an optional return value).

Reply:
For track color, it currently returns `RgbColor::optional_t`. So it shout actually return `std::optional<RgbColor::optional_t>`?

Reply:
Oh, you are right, I overlooked that `::optional_t` :see_no_evil:

Reply:
But if 0xFFFFFF is stored in the tags, the track color will be `RgbColor::nullopt`. So there's not really way to distinguish between an uncolored track and a track color that can't be imported. So maybe nesting optional makes sense?

Reply:
RgbColor::nullopt is now ignored on import.